docs/usage.rst

```diff
@@ -4,32 +4,40 @@
 Usage
 =====
 
-Create a :class:`~parsel.selector.Selector` object for the HTML or XML text
-that you want to parse::
+Create a :class:`~parsel.selector.Selector` object for your input text.
+
+For HTML or XML, use `CSS`_ or `XPath`_ expressions to select data::
 
     >>> from parsel import Selector
-    >>> text = "<html><body><h1>Hello, Parsel!</h1></body></html>"
-    >>> selector = Selector(text=text)
+    >>> html_text = "<html><body><h1>Hello, Parsel!</h1></body></html>"
+    >>> html_selector = Selector(text=html_text)
+    >>> html_selector.css('h1')
+    [<Selector query='descendant-or-self::h1' data='<h1>Hello, Parsel!</h1>'>]
+    >>> html_selector.xpath('//h1')  # the same, but now with XPath
+    [<Selector query='//h1' data='<h1>Hello, Parsel!</h1>'>]
 
-Then use `CSS`_ or `XPath`_ expressions to select elements::
+For JSON, use `JMESPath`_ expressions to select data::
 
-    >>> selector.css('h1')
-    [<Selector xpath='descendant-or-self::h1' data='<h1>Hello, Parsel!</h1>'>]
-    >>> selector.xpath('//h1')  # the same, but now with XPath
-    [<Selector xpath='//h1' data='<h1>Hello, Parsel!</h1>'>]
+    >>> json_text = '{"title":"Hello, Parsel!"}'
+    >>> json_selector = Selector(text=json_text)
+    >>> json_selector.jmespath('title')
+    [<Selector query='title' data='Hello, Parsel!'>]
 
 And extract data from those elements::
```

Comment:
Would it make sense to add json_selector extraction example as well? The flow breaks a bit now - we show xpath/css selectors, then show jmespath selector, and then say "extract data from these elements", with xpath/css again

Reply:
That was @felipeboffnunes’s plan.

My thinking is that the point of the `get` and `getall` examples is to showcase those. I am OK with adding a JSON example there, but I don’t think it is necessary. It may be better to, instead of adding, replacing one of the existing ones (XPath or CSS) with JMESPath.

We could also consider featuring `text()` here as a 3rd extraction method (related to https://github.com/scrapy/parsel/pull/243), to justify having 3 examples. But I still think the point of the examples of `get` and `getall` is to feature extraction methods, not selection methods, i.e. I don’t think we should keep extending those examples as we add more selection methods (https://github.com/scrapy/parsel/issues/204).

Reply:
I'm also fine with not having the example here, it's just that it reads a bit strange now - you have a json_selector example, then `And extract data from those elements::` caption, so I'd expect to get an example of data extraction from json_selector, based on this title. We may also move the json_selector section below extraction example, not sure.

Reply:
I was thinking that we could add a snippet of data with two elements on the json_text, this way we could display one HTML example with `get` and XPath (and leave CSS without an example, as it should follow Xpath's logic) and use `getall` on a json_selector that would bring more than one element.

Adding those elements to json_text would make the example more expressive, but also would increase the amount of information displayed...

A simpler version of this idea could be just having one `get` for Xpath and a `getall` for JSON (with the same snippets) and not having an example for CSS.

Reply:
+1 to the simpler version.